editor/plugins/gradient_editor_plugin.h

```diff
@@ -33,25 +33,106 @@
 
 #include "editor/editor_inspector.h"
 #include "editor/editor_plugin.h"
-#include "gradient_editor.h"
+#include "scene/resources/gradient_texture.h"
 
-class GradientReverseButton : public BaseButton {
-	GDCLASS(GradientReverseButton, BaseButton);
+class EditorSpinSlider;
+class ColorPicker;
+class PopupPanel;
+class GradientTexture1D;
 
-	int margin = 2;
+class GradientEdit : public Control {
+	GDCLASS(GradientEdit, Control);
 
+	Ref<Gradient> gradient;
+	Ref<GradientTexture1D> preview_texture;
+
+	PopupPanel *popup = nullptr;
+	ColorPicker *picker = nullptr;
+
+	bool snap_enabled = false;
+	float snap_count = 10;
```

Comment:
Should be an integer type (unless there is a reason for it which I overlooked)

Reply:
Thanks, I also noticed for some reason I made the hovered point's index a float. So fixed that too.